### PR TITLE
Add functionality for execution on AWS Batch

### PR DESCRIPTION
**Pull request by sminot**
This PR provides two feature enhancements:

1) The user can specify their jobRoleArn with the `aws.jobRoleArn` configuration setting. If specified, this will be added to the job definition used for execution on AWS Batch. This added feature provides useful functionality for users who manage permissions at the level of the job role (as opposed to the instance role) in multi-user environments.

2) The existing parameter `docker.temp` is now evaluated appropriately during execution on AWS Batch. When that variable is set in the configuration, a mount point is created during execution on AWS Batch. This allows the user to access arbitrary mountpoints for use as temporary storage, which can be a nice way to get a larger working directory.

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/Const.groovy [hunk 1]
```diff
@@ -55,12 +55,12 @@ class Const {
...
      * The app build time as linux/unix timestamp
      */
-    static public final long APP_TIMESTAMP = 1552584379498
+    static public final long APP_TIMESTAMP = 1552691843205
 
     /**
      * The app build number
      */
-    static public final int APP_BUILDNUM = 5061
+    static public final int APP_BUILDNUM = 5083
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/Const.groovy]: Please do not commit changes in Const file, since they are automatically generated.

**Comment by sminot** [comment]
This _should_ fix things, @pditommaso

**Comment by sivkovic** [comment]
Do you have any timeline for merging this pull request?

**Comment by pditommaso** [comment]
Hopefully soon, but there's no timeline.

**Comment by sminot** [comment]
Is there anything that we can do to help get this closer to being merged? Or is it being blocked by other development efforts?


---------------------------------------

Samuel Minot, PhD
Microbiome Research Initiative
Fred Hutchinson Cancer Research Center
sminot@fredhutch.org<mailto:sminot@fredhutch.org>
206-667-2884
https://minot.bio

On Apr 12, 2019, at 9:02 AM, Paolo Di Tommaso <notifications@github.com<mailto:notifications@github.com>> wrote:


Hopefully soon, but there's no timeline.

**Comment by pditommaso** [comment]
it's not a technical problem, I need to find time to include and to uniform
this change with the planned improvements for the Batch executor.

**Comment by pditommaso** [comment]
I've merged this manually with 651eeadd0852d66c0bafa8885dedc27214f36f33. 

Volumes and job role can be specified in the `aws.batch` scope, for example: 

```
  aws {
    region = 'eu-west-1'
    batch {
        jobRole = 'aws::arn::something/etc'
        volumes = ['/tmp', '/host/path:/mnt/path:ro']
    }
  }
```

**Comment by pditommaso** [comment]
cc @fstrozzi

**Pull request closed without merging**